TopMovies/Service/MovieService.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  MovieService.swift
+//  TopMovies
+//
+//  Created by anikolaenko on 02.02.2021.
+//
+
+import Foundation
+import ReSwift
+
+class MovieService: StoreSubscriber {
+    typealias StoreSubscriberStateType = MoviesListState
+
+    private let movieAPI: MovieAPIProtocol
+    
+    public init(movieAPI: MovieAPIProtocol) {
+        self.movieAPI = movieAPI
+    }
+    
+    func newState(state: MoviesListState) {
+        switch state.moviesList {
+        case .requested:
+            mainStore.dispatch(MoviesListAction.downloading)
+            movieAPI.topMovies { (result) in
+                switch result {
+                case let .success(data):
+                    mainStore.dispatch(MoviesListAction.completed(movies: data))
+                case let .failure(error):
+                    mainStore.dispatch(MoviesListAction.failed(error: error))
+                }
+            }
+        default: break
+        }
```

Comment:
Let's reduce that code and make it prettier by simply checking if `moviesList` is `requested`:
```swift
guard state.moviesList == .requested else { return }
//mainStore.dispatch...
//movieAPI.topMovies...
```

You gonna need `Equatable` conformance for state and model.